simple_certmanager/forms.py

```diff
@@ -52,6 +67,46 @@ def save(self, commit=True):
             instance.save()
         return instance
 
+    def clean(self) -> dict[str, Any]:
+        # Errors if the CSR already has a certificate
+        if self.instance.public_certificate:
+            self.add_error(
+                "certificate",
+                "A certificate already exists for this CSR. "
+                "Delete the certificate first.",
+            )
```

Comment:
I think this is a check you can put in `def clean_certificate`, it doesn't depend on other form fields

Reply:
this I did not manage to make it work. I wrote smth like :

```
    def clean_certificate(self) -> File | None:
        # Errors if the CSR already has a certificate
        if self.instance.public_certificate:
            self.add_error(
                "certificate",
                _(
                    "A certificate already exists for this CSR. "
                    "Delete the certificate first.",
                ),
            )
        return self.cleaned_data.get("certificate")
```

But then I've got empty files when uploading the cert.